web/src/features/datasets/server/dataset-router.ts

```diff
@@ -417,6 +431,27 @@ export const datasetRouter = createTRPCRouter({
           createdAt: "desc",
         },
       });
+
+      const observationIds = runItems.map((ri) => ri.observationId);
+      const observations = await ctx.prisma.observationView.findMany({
```

Comment:
this view is expensive, I'd only select the columns that we need to render the table, i.e. `calculatedTotalCost`

Reply:
I didn't now that the view is expensive.
Change is made 👍

Reply:
thank you. it is basically a collection of different observation related metrics. if needed, all can be used, but if we know that we only need a specific one, the query should be much faster